test/extended/router/h2spec.go

```diff
@@ -214,39 +214,54 @@ func failingTests(testSuites []*h2spec.JUnitTestSuite) []h2specFailingTest {
 	return failures
 }
 
-func runConformanceTests(oc *exutil.CLI, host, podName string) ([]*h2spec.JUnitTestSuite, error) {
-	g.By("Running the h2spec CLI test")
+func runConformanceTests(oc *exutil.CLI, host, podName string, timeout time.Duration) ([]*h2spec.JUnitTestSuite, error) {
+	var testSuites []*h2spec.JUnitTestSuite
 
-	// this is the output file in the pod
-	outputFile := "/tmp/h2spec-results"
+	if err := wait.Poll(time.Second, timeout, func() (bool, error) {
+		g.By("Running the h2spec CLI test")
 
-	// h2spec will exit with non-zero if _any_ test in the suite
-	// fails, or if there is a dial timeout, so we ignore the
-	// error. If we can fetch the results and if we can decode the
-	// results and we have > 0 test suites from the decoded
-	// results then assume the test ran.
-	output, _ := e2e.RunHostCmd(oc.Namespace(), podName, h2specCommand(h2specDialTimeoutInSeconds, host, outputFile))
+		// this is the output file in the pod
+		outputFile := "/tmp/h2spec-results"
 
-	g.By("Copying results")
-	data, err := e2e.RunHostCmd(oc.Namespace(), podName, fmt.Sprintf("cat %q", outputFile))
-	if err != nil {
-		return nil, err
-	}
-	if len(data) == 0 {
-		return nil, errors.New("zero length results file")
-	}
+		// h2spec will exit with non-zero if _any_ test in the suite
+		// fails, or if there is a dial timeout, so we log the
+		// error. But if we can fetch the results and if we can decode the
+		// results and we have > 0 test suites from the decoded
+		// results then assume the test ran.
+		output, err := e2e.RunHostCmd(oc.Namespace(), podName, h2specCommand(h2specDialTimeoutInSeconds, host, outputFile))
+		if err != nil {
+			e2e.Logf("error running h2spec: %v, but checking on result content", err)
```

Comment:
If there is an error, just try again.  Or if there is anything in the `output`, how about including `output` in the error log and start over?

Reply:
We don't `return false, nil`, because although the exit code may be non-zero the tests [should] have run. The steps to determine overall failure is an empty results file which is what seems to flake in CI. Does that answer your concern?